ui/perfherder/compare/CompareTableControls.jsx

```diff
@@ -27,34 +29,56 @@ export default class CompareTableControls extends React.Component {
       filteredText: this.getDefaultFilterText(this.validated),
       showRetriggerModal: false,
       currentRetriggerRow: {},
+      totalPagesList: 0,
+      page: this.validated.page ? parseInt(this.validated.page, 10) : 1,
+      countPages: 0,
     };
   }
 
   componentDidMount() {
     this.updateFilteredResults();
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps, prevState) {
     const { compareResults } = this.props;
+    const { countPages } = this.state;
+    const params = parseQueryParams(this.props.location.search);
+    const prevParams = parseQueryParams(prevProps.location.search);
+
+    if (prevState.countPages !== countPages) {
+      /* eslint-disable react/no-did-update-set-state */
+      this.setState({ totalPagesList: this.generatePages(countPages) });
+    }
     if (prevProps.compareResults !== compareResults) {
       this.updateFilteredResults();
     }
+    if (params.page && params.page !== prevParams.page) {
+      /* eslint-disable react/no-did-update-set-state */
```

Comment:
If you move the `/* eslint-disable react/no-did-update-set-state */` to the top of the file then you wont' need to write it twice.